### PR TITLE
Add decompositions for state preps

### PR DESCRIPTION
**Pull request by johannesjmeyer**
**Context:**
Recently, decompositions where introduced into Pennylane. Even more recently, decompositions of state preparations into gates where introduced as templates.

**Description of the Change:**
The state preparation templates `BasisStatePreparation` and `MottonenStatePreparation` will be used as decompositions for `BasisState` and `QubitStateVector`.

Additionally, the default argument `wires=None` was replaced with `wires` as `None` is not an acceptable value.

**Benefits:**
`BasisState` and `QubitStateVector` will be supported on hardware out-of-the-box.

**Possible Drawbacks:**
Users might use `QubitStateVector` on a device that does not support it and thus uses the decomposition and wonder why it takes so long.

### CODE REVIEW

File: pennylane/ops/qubit.py [hunk 1]
```diff
@@ -765,6 +765,13 @@ class QubitStateVector(Operation):
     * Number of parameters: 1
     * Gradient recipe: None
 
+    .. note::
+
+        If the ``BasisState`` operation is not supported natively on the
```
Comment (josh146) [inline pennylane/ops/qubit.py]:
```suggestion
        If the ``QubitStateVector`` operation is not supported natively on the
```

File: tests/test_operation.py [hunk 2]
```diff
@@ -794,3 +794,19 @@ def test_basis_state_decomposition(self):
...
+        with qml.utils.OperationRecorder() as rec1:
+            qml.QubitStateVector.decomposition(state, wires=[0, 1])
+
+        with qml.utils.OperationRecorder() as rec2:
+            qml.templates.state_preparations.MottonenStatePreparation(state, wires=[0, 1])
+
+        for op1, op2 in zip(rec1.queue, rec2.queue):
+            assert op1.name == op2.name
+            assert op1.wires == op2.wires
+            assert op1.params == op2.params
```
Comment (josh146) [inline tests/test_operation.py]:
```suggestion
            assert op1.params == op2.params

```

Comment (josh146) [reply tests/test_operation.py]: Is it worth also adding an integration test here? For example, for a mocked default.qubit with no `QubitStateVector`, and ensure that the prepared state is correct?

Comment (johannesjmeyer) [reply tests/test_operation.py]: Actually I think now that just testing that `BasisStateDecomposition` and `MottonenStatePreparation` are called is wholly sufficient. The state preparation templates and the logic of decomposing when the operation is not supported are properly tested elsewhere.

**Comment by johannesjmeyer** [comment]
I just changed the tests of the new decompositions to only test that the correct underlying template is called.

File: tests/test_default_gaussian.py [hunk 3]
```diff
@@ -625,7 +625,7 @@ def test_unsupported_gates(self, g, gaussian_device_3_wires):
         """Test error is raised with unsupported gates"""
         # TODO: Refactor this test to remove dynamic attribute loading, and instead
         # have separate tests for gates with specific number of parameters/wires.
-        if g == "BasisState":
+        if g == "BasisState" or g == "QubitStateVector":
             pytest.skip("Test not set up properly for gates with array parameters, needs refactoring.")
```
Comment (josh146) [inline tests/test_default_gaussian.py]: We definitely need to refactor this test soon 😟

Comment (johannesjmeyer) [reply tests/test_default_gaussian.py]: Totally! Was weird to introduce a new decomposition for a qubit op and then have to refactor `test_default_gaussian.py`...

**Pull request merged**